pipelines/zendesk/helpers/talk_api.py

```diff
@@ -0,0 +1,115 @@
+"""
+This module contains everything related to the API client class made to make requests specifically to ZendeskTalk
+"""
+from pipelines.zendesk.helpers.credentials import ZendeskCredentialsToken, ZendeskCredentialsOAuth, ZendeskCredentialsEmailPass
+from dlt.common import logger
+from dlt.common.typing import DictStrStr, TSecretValue, TDataItems
+from pendulum import DateTime
+from time import sleep
+from typing import Union, Tuple, Iterator
+import requests
+
+
+TALK_ENDPOINTS = {
+    "calls": "/api/v2/channels/voice/calls",
+    "addresses": "/api/v2/channels/voice/addresses",
+    "greeting_categories": "/api/v2/channels/voice/greeting_categories",
+    "greetings": "/api/v2/channels/voice/greetings",
+    "ivrs": "/api/v2/channels/voice/ivr",
+    "phone_numbers": "/api/v2/channels/voice/phone_numbers",
+    "settings": "/api/v2/channels/voice/settings",
+    "lines": "/api/v2/channels/voice/lines",
+    "agents_activity": "/api/v2/channels/voice/stats/agents_activity",
+    "current_queue_activity": "/api/v2/channels/voice/stats/current_queue_activity"
+}
+possible_endpoints = {
+    "incremental_calls": "/api/v2/channels/voice/stats/incremental/calls.json",
+    "incremental_legs": "/api/v2/channels/voice/stats/incremental/legs.json",
+    "availabilities": "/api/v2/channels/voice/availabilities",
+    "recordings": "/api/v2/channels/voice/calls/{call_id}/recordings",
+    "digital_lines": "/api/v2/channels/voice/digital_lines",
+    "agents_overview": "/api/v2/channels/voice/stats/agents_overview",
+    "account_overview": "/api/v2/channels/voice/stats/account_overview",
+}
+ZENDESK_STATUS_CODES = {
+    "ok": 200,
+    "rate_limit": 429
+}
+
+
+class ZendeskAPIClient:
+    """
+    API client used to make requests to Zendesk TALK.
+    """
+    subdomain: str = ""
+    url: str = ""
+    headers: DictStrStr = {}
+    auth: Tuple[str, TSecretValue]
+
+    def __init__(self, credentials: Union[ZendeskCredentialsOAuth, ZendeskCredentialsToken, ZendeskCredentialsEmailPass]) -> None:
+        """
+        Initializer for the API client which is then used to make API calls to the ZendeskAPI
+        @:param credentials: ZendeskCredentials object which contains the necessary credentials to authenticate to ZendeskAPI
+        @:returns: Nothing since it just initializes API parameters
+        """
+        # set subdomain, this is always needed to configure endpoints
+        self.subdomain = credentials.subdomain
```

Comment:
what was the problem with using `zenpy` here? we basically write same auth code twice. can't we use this client instead of zenpy then?

Reply:
ZendeskTalk is not functional in Zenpy despite part of it showing up as implemented.